pkg/store/driver/txn/unionstore_driver.go

```diff
@@ -67,20 +68,30 @@ func (m *memBuffer) GetFlags(key kv.Key) (kv.KeyFlags, error) {
 }
 
 func (m *memBuffer) Staging() kv.StagingHandle {
+	if m.isPipelinedDML {
+		// 0 stands for staging not supported.
+		return 0
```

Comment:
We should panic in these methods to avoid incorrect operations.

Reply:
And add checks before every usage of these methods.

Reply:
I did this follow BR's implemenration and the comment.

https://github.com/pingcap/tidb/blob/ac057ebbc37c7dfee2df1d31a43d18db1cbf672a/br/pkg/lightning/backend/kv/session.go#L174-L177

https://github.com/pingcap/tidb/blob/ac057ebbc37c7dfee2df1d31a43d18db1cbf672a/pkg/table/tables/mutation_checker.go#L86-L89

If we want to panic in those case, better to add a method `SupportStaging` and avoid calling `Staging` and `Release` and `Cleanup` for BR also.

Reply:
Okay it's not a big problem

Reply:
Is it problematic that we do not check usages of these methods in tables.go?

Reply:
Yes, a little obscure because it's actually no effects and no error returned.